rest.go

```diff
@@ -0,0 +1,148 @@
+// Package rest allows for quick and easy access any REST or REST-like API.
+package rest
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// Version represents the current version of the rest library
+const Version = "0.1"
+
+// Method contains the supported HTTP verbs.
+type Method string
+
+// Supported HTTP verbs.
+const (
+	Get    Method = "GET"
+	Post   Method = "POST"
+	Put    Method = "PUT"
+	Patch  Method = "PATCH"
+	Delete Method = "DELETE"
+)
+
+// Request holds the request to an API Call.
+type Request struct {
+	Method      Method
+	BaseURL     string // e.g. https://api.mayadata.com
+	Headers     map[string]string
+	QueryParams map[string]string
+	Body        []byte
+}
+
+// RestError is a struct for an error handling.
+type RestError struct {
+	Response *Response
+}
+
+// Error is the implementation of the error interface.
+func (e *RestError) Error() string {
+	return e.Response.Body
+}
+
+// DefaultClient is used if no custom HTTP client is defined
+var DefaultClient = &Client{HTTPClient: http.DefaultClient}
+
+// Client allows modification of client headers, redirect policy
+// and other settings
+// See https://golang.org/pkg/net/http
+type Client struct {
+	HTTPClient *http.Client
+}
+
+// Response holds the response from an API call.
+type Response struct {
+	StatusCode int                 // e.g. 200
+	Body       string              // e.g. {"result: success"}
+	Headers    map[string][]string // e.g. map[X-Ratelimit-Limit:[600]]
+}
+
+// AddQueryParameters adds query parameters to the URL.
+func AddQueryParameters(baseURL string, queryParams map[string]string) string {
+	baseURL += "?"
+	params := url.Values{}
```

Comment:
What is url?
Have we run unit test?
Can we include the Dockerfile to run the unit test & run it in Github actions